akka-actor/src/main/scala/akka/pattern/HandleBackoff.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.pattern
+
+import akka.actor.{ Actor, ActorRef, Props }
+
+private[akka] trait HandleBackoff {
+  this: Actor ⇒
+  def childProps: Props
+  def childName: String
+  def reset: BackoffReset
+  protected def handleMessageToChild(m: Any): Unit
+
+  var child: Option[ActorRef] = None
+  var restartCount = 0
+  var finalStopMessageReceived = false
+
+  import BackoffSupervisor._
+  import context.dispatcher
+
+  override def preStart(): Unit = startChild()
+
+  def startChild(): Unit = {
+    if (child.isEmpty) {
+      child = Some(context.watch(context.actorOf(childProps, childName)))
+    }
+  }
+
+  def handleBackoff: Receive = {
+    case StartChild ⇒
+      startChild()
+      reset match {
+        case AutoReset(resetBackoff) ⇒
+          val _ = context.system.scheduler.scheduleOnce(resetBackoff, self, ResetRestartCount(restartCount))
```

Comment:
True, I've just moved HandleBackoff into a separate file (from BackoffSupervisor) since it's used by both backoff supervisor. My only change to this trait was extracting the msg fallback handling into a method.

I will amend this.